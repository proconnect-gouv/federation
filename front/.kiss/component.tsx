import React from 'react';

interface DumbComponentProps {
  className: string;
}

export const DumbComponent: React.FC<DumbComponentProps> = React.memo(
  ({ className }: DumbComponentProps) => {
    return <div />;
  },
);

DumbComponent.displayName = 'DumbComponent';
