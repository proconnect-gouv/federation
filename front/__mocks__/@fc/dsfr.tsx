export const FranceConnectButton = jest.fn(() => <div>FranceConnectButton</div>);

export const SimpleButton = jest.fn(() => <div>SimpleButton</div>);

export const CheckboxInput = jest.fn(() => <div>CheckboxInput</div>);

export const ToggleInput = jest.fn(() => <div>ToggleInput</div>);

export const ApplicationLayout = jest.fn(() => <div>ApplicationLayout</div>);

export const LinkComponent = jest.fn(() => <div>LinkComponent</div>);

export const SearchBarComponent = jest.fn(() => <div>SearchBarComponent</div>);

export const BadgeComponent = jest.fn(() => <div>BadgeComponent</div>);

export const AlertComponent = jest.fn(({ children }) => (
  <div data-mockid={'AlertComponent'}>
    <div>AlertComponent</div>
    <div>{children}</div>
  </div>
));

export const PaginationComponent = jest.fn(() => <div>PaginationComponent</div>);

export const AccordionComponent = jest.fn(({ children }) => (
  <div data-mockid={'AccordionComponent'}>
    <h3>AccordionComponent</h3>
    <div>{children}</div>
  </div>
));

export const AccordionGroupComponent = jest.fn(({ children }) => (
  <div data-mockid={'AccordionGroupComponent'}>
    <h3>AccordionGroupComponent</h3>
    <div>{children}</div>
  </div>
));

export const Breakpoints = {
  LG: 1248,
  MD: 992,
  SM: 768,
  XS: 576,
};

export const Sizes = {
  LARGE: 'lg',
  MEDIUM: 'md',
  SMALL: 'sm',
};

export const AlertTypes = {
  ERROR: 'error',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
};

export const IconPlacement = {
  LEFT: 'left',
  RIGHT: 'right',
};

export const Priorities = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',
};

export const ButtonTypes = {
  BUTTON: 'button',
  RESET: 'reset',
  SUBMIT: 'submit',
};
