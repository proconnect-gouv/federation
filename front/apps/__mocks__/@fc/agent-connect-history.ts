export const useUserHistory = jest.fn(() => [
  { active: false, display: true, name: 'mock-name-1', uid: 'mokc-uid-1' },
  { active: false, display: true, name: 'mock-name-2', uid: 'mokc-uid-2' },
  { active: false, display: true, name: 'mock-name-3', uid: 'mokc-uid-3' },
  { active: false, display: true, name: 'mock-name-4', uid: 'mokc-uid-4' },
]);

export const useAddToUserHistory = jest.fn(() => jest.fn());

export const useRemoveFromUserHistory = jest.fn(() => jest.fn());
