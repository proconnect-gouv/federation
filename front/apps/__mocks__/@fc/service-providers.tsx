export const ServiceProviderItemComponent = jest.fn(({ datapassId }) => (
  <div>ServiceProviderItemComponent {datapassId}</div>
));

export {
  ServiceProviderEditActionTypes,
  ServiceProvidersActionTypes,
  ServiceProvidersPermissionTypes,
  ServiceProviderViewActionTypes,
} from '@fc/service-providers';
