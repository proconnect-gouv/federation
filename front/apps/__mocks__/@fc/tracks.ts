export const TracksListComponent = jest.fn();
