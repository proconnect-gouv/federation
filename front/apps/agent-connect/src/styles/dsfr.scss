// @TODO a supprimer une fois la lib dsfr partag√©e entre les apps front

.mr8 {
  margin-right: 8px;
}

.is-block {
  display: block;
}

.flex-columns {
  display: flex;
  flex-direction: row;
}

.items-center {
  align-items: center;
}

a.no-underline {
  text-decoration: none;
}

.bg-blue-france-100 {
  background-color: #f2f2f9;
}

.is-blue-agentconnect {
  color: #0579ee;
}

.fr-display-xs {
  font-size: 2.5rem;
  line-height: 3rem;
}

h1 {
  font-size: 2rem;
  line-height: 2.5rem;
}

.content-wrapper-sm {
  margin-left: auto;
  margin-right: auto;
}

.my32 {
  margin-bottom: 2rem;
  margin-top: 2rem;
}

.mb32 {
  margin-bottom: 2rem;
}

.px120 {
  padding-left: 7.5rem;
  padding-right: 7.5rem;
}

.fr-text-lg {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.p24 {
  padding: 1.5rem;
}

.no-no-padding {
  padding: 0 !important;
}

@media (min-width: 768px) {
  .fr-display-xs {
    font-size: 3rem;
    line-height: 3.5rem;
  }

  h1 {
    font-size: 2.5rem;
    line-height: 3rem;
  }

  .content-wrapper-sm {
    max-width: 792px;
  }

  .fr-text-lg {
    font-size: 1.25rem;
    line-height: 2rem;
  }
}
