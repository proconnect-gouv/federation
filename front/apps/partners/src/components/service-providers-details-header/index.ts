/* istanbul ignore file */

// declarative file
export * from './service-providers-details-header.component';
