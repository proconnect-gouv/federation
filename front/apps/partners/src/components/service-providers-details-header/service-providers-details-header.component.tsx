import React from 'react';
import { Link } from 'react-router-dom';

import { BadgeComponent, Sizes } from '@fc/dsfr';
import { t } from '@fc/i18n';

interface ServiceProvidersDetailsHeaderProps {
  item: {
    color: string;
    platformName: string;
    status: string;
    spName: string;
  };
}

export const ServiceProvidersDetailsHeaderComponent = React.memo(
  ({ item }: ServiceProvidersDetailsHeaderProps) => {
    const { color, platformName, spName, status } = item;
    return (
      <React.Fragment>
        <div className="fr-mb-2w">
          <Link
            className="fr-btn fr-btn--tertiary-no-outline fr-fi-arrow-go-back-fill fr-btn--icon-left"
            data-testid="ServiceProvidersDetailsHeaderComponent-return-sp-list-button"
            to="/service-providers">
            {t('ServiceProvidersDetailsPage.returnButton')}
          </Link>
        </div>
        {/* @TODO: add classname to BadgeComponent to match with designs spacing */}
        <BadgeComponent
          noIcon
          colorName={color}
          dataTestId="ServiceProvidersDetailsHeaderComponent-badge"
          label={status}
          size={Sizes.MEDIUM}
        />
        <span className="fr-ml-2w" data-testid="ServiceProvidersDetailsHeaderComponent-platform">
          {platformName}
        </span>
        <h1
          className="fr-mt-1w text-left"
          data-testid="ServiceProvidersDetailsHeaderComponent-spName">
          {spName}
        </h1>
      </React.Fragment>
    );
  },
);

ServiceProvidersDetailsHeaderComponent.displayName = 'ServiceProvidersDetailsHeaderComponent';
