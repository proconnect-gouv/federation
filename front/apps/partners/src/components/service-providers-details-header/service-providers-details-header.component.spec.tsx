import { render } from '@testing-library/react';

import { BadgeComponent } from '@fc/dsfr';

import { ServiceProvidersDetailsHeaderComponent } from './service-providers-details-header.component';

jest.mock('@fc/dsfr');
describe('ServiceProvidersDetailsHeaderComponent', () => {
  const ServiceProviderMock = {
    color: 'yellow-tournesol',
    platformName: 'FranceConnect',
    spName: 'Service Provider - Sandbox',
    status: 'en intÃ©gration',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    // when
    const { container } = render(
      <ServiceProvidersDetailsHeaderComponent item={ServiceProviderMock} />,
    );

    // then
    expect(container).toMatchSnapshot();
  });

  it('should display service provider name', () => {
    // when
    const { getByText } = render(
      <ServiceProvidersDetailsHeaderComponent item={ServiceProviderMock} />,
    );

    // then
    const element = getByText(`${ServiceProviderMock.spName}`);
    expect(element).toBeInTheDocument();
  });

  it('should display service provider platform', () => {
    // when
    const { getByText } = render(
      <ServiceProvidersDetailsHeaderComponent item={ServiceProviderMock} />,
    );

    // then
    const element = getByText(ServiceProviderMock.platformName);
    expect(element).toBeInTheDocument();
  });

  it('should call BadgeComponent', () => {
    // when
    render(<ServiceProvidersDetailsHeaderComponent item={ServiceProviderMock} />);

    // then
    expect(BadgeComponent).toHaveBeenCalled();
  });

  it('should call BadgeComponent with props', () => {
    // when
    render(<ServiceProvidersDetailsHeaderComponent item={ServiceProviderMock} />);

    // then
    expect(BadgeComponent).toHaveBeenCalledTimes(1);
    expect(BadgeComponent).toHaveBeenCalledWith(
      {
        colorName: ServiceProviderMock.color,
        dataTestId: 'ServiceProvidersDetailsHeaderComponent-badge',
        label: ServiceProviderMock.status,
        noIcon: true,
        size: 'md',
      },
      {},
    );
  });
});
