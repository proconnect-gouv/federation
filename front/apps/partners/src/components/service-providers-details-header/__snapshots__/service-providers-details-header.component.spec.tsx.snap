// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceProvidersDetailsHeaderComponent should match the snapshot 1`] = `
<div>
  <div
    class="fr-mb-2w"
  >
    <span>
      ServiceProvidersDetailsPage.returnButton
    </span>
  </div>
  <div>
    BadgeComponent
  </div>
  <span
    class="fr-ml-2w"
    data-testid="ServiceProvidersDetailsHeaderComponent-platform"
  >
    FranceConnect
  </span>
  <h1
    class="fr-mt-1w text-left"
    data-testid="ServiceProvidersDetailsHeaderComponent-spName"
  >
    Service Provider - Sandbox
  </h1>
</div>
`;
